include/lbann/optimizers/sgd.hpp

```diff
@@ -73,7 +73,9 @@ class sgd : public Cloneable<sgd<TensorDataType>,
   /** Archive for checkpoint and restart */
   template <class Archive> void serialize(Archive & ar) {
     ar(cereal::base_class<data_type_optimizer<TensorDataType>>(this),
-       CEREAL_NVP(m_momentum));
+       CEREAL_NVP(m_momentum),
+       CEREAL_NVP(m_nesterov));
+    // CEREAL_NVP(m_velocity));
```

Comment:
The optimizer state should be serialized, or else we'll se wonky learning after deserializing.

Reply:
Do you mean that `m_velocity` should be serialized? Or are you referring to something else? If you're referring to `m_velocity`, this is the issue we discussed yesterday. Distributed matrices cannot yet be serialized inside LBANN classes. The reason for this is that the LBANN load functions will read an XML archive on rank 0, pipe it to a binary archive, broadcast the archive to the rest of the trainer, and then restore from binary. This will change as we continue the refactor, so I have just left these here as comments so it's easy to restore them in the next PR that will address a wider view of the "checkpoint workflow", if you will.